File: find-max-divisors/findMaxDivisors.js
const findMaxDivisors = (a, b) => {
    if(b - a === 0) return a;
    if(b - a === 1){
        if(a % 2 === 0 === true) return a;
        if(b % 2 === 0 === true) return b;
    };

    let start = a
    let end = b;
    if(a % 2 === 0 === false) start += 1;
    if(b % 2 === 0 === false) end -= 1;
    
    let factors = 0;
    const divisors = {};
    for(let i = start; i < end; i++){
        for(let j = 1; j <= Math.sqrt(i); j++){
            if(i % j === 0) factors++;
            if(j !== i / j) factors++;
        };
        Object.assign(divisors, {[i]: factors});
        factors = 0;
    };

    const output = [];
    const maxDivisors = Math.max(... Object.values(divisors));
    for(let key in divisors){
        if(divisors[key] === maxDivisors){
            output.push(start);
        };
        start++;
    };
    console.log(divisors, output)
    return output.sort((a, b) => {
        return a - b;
    })[0];

};

module.exports = { findMaxDivisors };

File: python-venv/find_max_divisors/kata_function/find_max_divisors.py
import math

def find_max_divisors(a, b):
    if b - a == 0:
        return a
    if b - a == 1:
        check = [a, b]
        return max(check)

    start = a
    factors = 0
    divisors = {}
    for i in range(start, b + 1, 1):
        for j in range(1, round(math.sqrt(i)) + 1, 1): 
            if i % j == 0:
                factors += 1
                if j != i / j:
                    factors += 1
        divisors[i] = factors
        factors = 0

    maxDivisors = max(divisors.values())
    for key in divisors:
        if divisors[key] == maxDivisors:
            return start
        start += 1


#https://www.codewars.com/kata/588c0a38b7cd14085300003f/train/python